src/coreclr/vm/gcinfodecoder.cpp

```diff
@@ -1488,18 +1488,26 @@ template <typename GcInfoEncoding> OBJECTREF* TGcInfoDecoder<GcInfoEncoding>::Ge
                         PREGDISPLAY     pRD
                         )
 {
+#if defined(TARGET_UNIX)
+    _ASSERTE(regNum >= 0 && regNum <= 32);
+#else // TARGET_UNIX
     _ASSERTE(regNum >= 0 && regNum <= 16);
+#endif // TARGET_UNIX
     _ASSERTE(regNum != 4);  // rsp
 
 #ifdef FEATURE_NATIVEAOT
     PTR_uintptr_t* ppRax = &pRD->pRax;
     if (regNum > 4) regNum--; // rsp is skipped in NativeAOT RegDisplay
-#else
+#else // FEATURE_NATIVEAOT
     // The fields of KNONVOLATILE_CONTEXT_POINTERS are in the same order as
     // the processor encoding numbers.
-
     ULONGLONG **ppRax = &pRD->pCurrentContextPointers->Rax;
-#endif
+    if(ExecutionManager::GetEEJitManager()->IsAPXSupported() && regNum >= 16)
```

Comment:
When APX is not supported, we should never see `regNum >= 16` here. It is sufficient and more efficient to just check `regNum >= 16`. 

`IsAPXSupported` can be assert.

Also, GCInfoDecoder is used in number of cross-process diagnostic/debugger scenarios (see https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/gcinfodecoder.h#L13). If this code was dependent on checking IsAPXSupported, you would have to figure out how to get this value for the target.